### PR TITLE
htop: optionally enable sensors and systemd support on Linux

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
htop tries to dynamically load these deps and (obviously) fails. We can manually
add them to the rpath to fix the issue.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nagy** [comment]
I can confirm that the systemd meter now works with this PR.

Thanks for pointing out this feature. Really useful.

**Review by aanderse — APPROVED** [review]
❤️

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 134583` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>htop</li>
  </ul>
</details>

**Comment by davidak** [comment]
very nice

Bildschirmfoto von 2021-08-19 01-29-25

**Review by davidak — APPROVED** [review]
Change looks good

Builds, works

File: pkgs/tools/system/htop/default.nix [hunk 1]
```diff
@@ -1,7 +1,14 @@
 { lib, fetchFromGitHub, stdenv, autoreconfHook
-, ncurses, IOKit
+, ncurses
+, IOKit
+, sensorsSupport ? stdenv.isLinux, lm_sensors
+, systemdSupport ? stdenv.isLinux, systemd
 }:
 
+with lib;
+
```
Comment (SuperSandro2000) [inline pkgs/tools/system/htop/default.nix:9]: lib. is so nice and cute, we don't need to use a with over the entire file.

Comment (davidak) [reply pkgs/tools/system/htop/default.nix:9]: correct, but off-topic in this pr

**Pull request merged**